lib/nerves_hub_user_api/product_user.ex

```diff
@@ -0,0 +1,84 @@
+defmodule NervesHubUserAPI.ProductUser do
+  @moduledoc """
+  Manage ProductUsers on NervesHub
+
+  Path: /orgs/:org_name/products/:product_name/users
+  """
+
+  @type role :: :admin | :delete | :write | :read
```

Comment:
The `role` type seems like it would fit better in `nerves_hub_user_api.ex` (open to other ideas) plus you could `defguard` it to get ride of the role check duplication.